src/main/java/org/jabref/logic/cleanup/ConvertToBiblatexCleanup.java

```diff
@@ -39,6 +39,12 @@
                 entry.clearField(StandardField.YEAR).ifPresent(changes::add);
                 entry.clearField(StandardField.MONTH).ifPresent(changes::add);
             });
+        } else {
+            // If field date is filled it should be removed the year and month fields
+            entry.getFieldOrAlias(StandardField.DATE).ifPresent(newDate -> {
+                entry.clearField(StandardField.YEAR).ifPresent(changes::add);
```

Comment:
I think there should also be a check that the information in the date field coincides with the one in year & month. For example, if `date = {2004-05}` but `year = {2005}` the year field shouldn't be deleted.

Reply:
Is this really the job of this class? The code assures, that there is something in the date field. I don't think that the cleanup class has to check, if the conversion was done right...